pkg/resources/materialized_view_acceptance_test.go

```diff
@@ -0,0 +1,100 @@
+package resources_test
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+)
+
+func TestAcc_MaterializedView(t *testing.T) {
+	dbName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	schemaName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	tableName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	viewName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+
+	resource.ParallelTest(t, resource.TestCase{
+		Providers: providers(),
+		Steps: []resource.TestStep{
+			{
+				Config: materializedViewConfig(dbName, schemaName, tableName, viewName, []string{"ID", "DATA"}, fmt.Sprintf("SELECT ID, DATA FROM %s.%s;", schemaName, tableName)),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "name", viewName),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "database", dbName),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "schema", schemaName),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "comment", "Terraform test resource"),
+					resource.TestCheckResourceAttr("snowflake_materialized_view.test", "cluster_by", "[\"ID\",\"DATA\"]"),
+					checkBool("snowflake_materialized_view.test", "is_secure", true),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_MaterializedView2(t *testing.T) {
+	dbName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	schemaName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	tableName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+	viewName := strings.ToUpper(acctest.RandStringFromCharSet(10, acctest.CharSetAlpha))
+
+	resource.ParallelTest(t, resource.TestCase{
+		Providers: providers(),
+		Steps: []resource.TestStep{
+			{
+				Config: materializedViewConfig(dbName, schemaName, tableName, viewName, []string{"ID", "DATA"}, fmt.Sprintf("SELECT ID, DATA FROM %s.%s WHERE ID LIKE 'foo%%';", schemaName, tableName)),
```

Comment:
I _think_ you need to specify the database as well. I'm not sure though. It's inconsistent here with your other tests.

Reply:
Hi, I confirmed that database and schema is not required if table is in the same database and schema.